pvlib/atmosphere.py

```diff
@@ -698,3 +698,153 @@ def angstrom_alpha(aod1, lambda1, aod2, lambda2):
+        with broadband irradiance reaching a module's cells to estimate
+        effective irradiance, i.e., the irradiance that is converted to
+        electrical current.
+
+    References
+    ----------
+    .. [1] Caballero, J.A., Fernández, E., Theristis, M.,
+        Almonacid, F., and Nofuentes, G. "Spectral Corrections Based on
+        Air Mass, Aerosol Optical Depth and Precipitable Water
+        for PV Performance Modeling."
+        IEEE Journal of Photovoltaics 2018, 8(2), 552-558.
+        https://doi.org/10.1109/jphotov.2017.2787019
+    .. [2] Gueymard, Christian. SMARTS2: a simple model of the
+        atmospheric radiative transfer of sunshine: algorithms
+        and performance assessment. Cocoa, FL:
+        Florida Solar Energy Center, 1995.
+    .. [3] Theristis, M., Fernández, E., Almonacid, F., and
+       Pérez-Higueras, Pedro. "Spectral Corrections Based
+       on Air Mass, Aerosol Optical Depth and Precipitable
+       Water for CPV Performance Modeling."
+       IEEE Journal of Photovoltaics 2016, 6(6), 1598-1604.
+       https://doi.org/10.1109/jphotov.2016.2606702
+
+        """
+
+    # Experimental coefficients
+
+    _coefficients = {}
+    _coefficients['cdte'] = (
+        1.0044, 0.0095, -0.0037, 0.0002, 0.0000, -0.0046,
+        -0.0182, 0, 0.0095, 0.0068, 0, 1)
+    _coefficients['monosi'] = (
+        0.9706, 0.0377, -0.0123, 0.0025, -0.0002, 0.0159,
+        -0.0165, 0, -0.0016, -0.0027, 1, 0)
+    _coefficients['multisi'] = (
+        0.9836, 0.0254, -0.0085, 0.0016, -0.0001, 0.0094,
+        -0.0132, 0, -0.0002, -0.0011, 1, 0)
+    _coefficients['cigs'] = (
+        0.9801, 0.0283, -0.0092, 0.0019, -0.0001, 0.0117,
+        -0.0126, 0, -0.0011, -0.0019, 1, 0)
+    _coefficients['asi'] = (
+        1.1060, -0.0848, 0.0302, -0.0076, 0.0006, -0.1283,
+        0.0986, -0.0254, 0.0156, 0.0146, 1, 0)
+    _coefficients['perovskite'] = (
+        1.0637, -0.0491, 0.0180, -0.0047, 0.0004, -0.0773,
+        0.0583, -0.0159, 0.01251, 0.0109, 1, 0)
+
+    if module_type is None and coefficients is None:
+        raise ValueError('Must provide either `module_type` or `coefficients`')
+    if module_type is not None and coefficients is not None:
+        raise ValueError('Only one of `module_type` and `coefficients` should '
+                         'be provided')
+    if module_type is not None:
+        coeff = _coefficients[module_type]
+    else:
+        coeff = coefficients
+
+    # Evaluate Spectral Shift
+    ama = airmass_absolute
+    modifier = (
+        coeff[0] + (ama) * coeff[1] + (ama * ama) * coeff[2]
+        + (ama * ama * ama) * coeff[3] + (ama * ama * ama * ama) * coeff[4]
+        + (aod500 - aod500_ref) * coeff[5]
+        + ((aod500 - aod500_ref) * (ama) * coeff[6]) * coeff[10]
+        + ((aod500 - aod500_ref) * (np.log(ama)) * coeff[6]) * coeff[11]
+        + (aod500 - aod500_ref) * (ama * ama) * coeff[7]
+        + (pw - pw_ref) * coeff[8] + (pw - pw_ref) * (np.log(ama)) * coeff[9])
```

Comment:
This calculation is neither easy to read nor efficient.  I suggest you rewrite in three (like in the paper) efficient subcalculations and return their sum.

Also add code comments to explain the extra coefficients.

Reply:
Hi @adriesse , this is the way I know to calculate the spectral modifier index. Please, feel free to propose another set of equations to do it and we can commit them. Thanks!

Reply:
Ok, we can add code comments to explain the extra coefficients. The reason to add it was to unify the number of coefficients to be applied for each photovoltaic technology. Then, the extra coefficient is set at 0 or 1 to take into account the coefficient when needed.